D1673/P1673.bs

```diff
@@ -5088,61 +5064,77 @@ satisfy the following requirements.
-
-4. The algorithm may assign arbitrarily many times to any reference
-    resulting from a valid output `mdspan` access,
-    as long as the algorithm's asymptotic complexity requirements are satisfied.
-
-5. The algorithm may assign any addend
-    in the algorithm's mathematical expression
-    to any reference resulting from a valid output `mdspan` access.
-
-6. The algorithm may reorder addends in its mathematical expression arbitrarily.
+1. Any input or output `mdspan`'s `value_type`
+    meets the requirements of `semiregular`.
+
+2. The algorithm may perform read-only access
+    on any input or output `mdspan` arbitrarily many times.
+
+3. The algorithm or method may make arbitrary many objects
+    of any input or output `mdspan`'s `value_type`,
+    value-initializing them or direct-initializing them
+    with any existing object of that type.
+
+4. The algorithm may assign arbitrarily many times
+    to any reference
+    resulting from a valid output `mdspan` access.
+
+5. If the algorithm's mathematical expression
+    uses addition,
+    but not subtraction, multiplication, or division,
+    then the algorithm evaluates the mathematical expression
+    using a sequence of evaluations of
+    `+`, `+=`, and `=` operators
+    that would produce the correct result
+    when operating on elements of a commutative semigroup.
+
+6. If the algorithm's mathematical expression
+    uses multiplication and possibly also addition,
+    but not subtraction or division,
+    then the algorithm
+    evaluates the mathematical expression
+    using a sequence of evaluations of
+    `*`, `*=`, `+`, `+=`, and `=` operators
+    that would produce the correct result
+    when operating on elements of a semiring
+    with noncommutative multiplication.
+
+7. If the algorithm's mathematical expression
+    uses subtraction and possibly also
+    addition and multiplication, but not division,
+    then the algorithm
+    evaluates the mathematical expression
+    using a sequence of evaluations of
+    `*`, `*=`, `+`, `+=`,
+    unary `-`, binary `-`, `-=`,
+    and `=` operators
+    that would produce the correct result
+    when operating on elements of a ring
+    with noncommutative multiplication.
+
+8. If the algorithm's mathematical expression
+    uses division and possibly also
+    addition, subtraction, and multiplication,
+    then the algorithm
+    evaluates the mathematical expression
+    using a sequence of evaluations of
+    `*`, `*=`, `/`, `/=`,
+    `+`, `+=`, unary `-`, binary `-`, `-=`,
+    and `=` operators
+    that would produce the correct result
+    when operating on elements of a field
+    with noncommutative multiplication.
+
+9. The algorithm may reorder addends in its mathematical expression arbitrarily.
     *[Note:*
-    The algorithm may *not* reorder products arbitrarily.
+    Factors in each product are not reordered.
     This is because the algorithms do not assume that multiplication is commutative.
```

Comment:
I'd omit this or reduce it to a suffix of the previous sentence:

> ; multiplication need not be commutative.

Reply:
Fixed; thanks!